src/webextension/ticker-timer-modes.js

```diff
@@ -37,27 +37,30 @@ function set_listeners_for_timer_mode(mode) {
     }
 };
 
-async function get_current_url_default() {
-    // returns a promise that resolves to the url of the active window/tab
+async function get_current_url_internal(mode) {
     try {
-        let tabs = await browser.tabs.query({currentWindow: true, active: true});
-        return new URL(tabs[0].url);
-
+        let tabs = await browser.tabs.query({currentWindow: true, active: true}),
+            tab = tabs[0];
+        if (tab.incognito) {
+            // mainly visual confirmation ("we are not logging")
+            // (but also, exception in popup: getBackgroundPage() returns undefined)
+            browser.browserAction.disable(tab.id);
+            // hide badge for this tab (do not reflect global mode changes from other windows, etc.)
+            browser.browserAction.setBadgeText({ text: "", tabId: tab.id });
+            return null;
```

Comment:
I don't think the badge text should be set in this function.  What if we return another special url (like the blue-mode url) for incognito tabs.  Then we can handle updating the badge based on that url.  Maybe the badge text should be "x" or "-" for private windows.

Reply:
The main reason to set it in *this* function was that I needed the tab id. I found it very irritating to watch the non-incognito window in the background make the same change (or vice-versa, the incognito window suddenly reflects the state of another window).

I actually tried the "-" and it looked okay, but empty+disabled was much more confidence-inspiring that it's really off. I think "-" or "---" will work if the browserAction is not disabled (e.g. the popup has some explanation).